gtfs/spec/en/reference.md

```diff
@@ -343,11 +345,29 @@ For examples that demonstrate how to specify a fare structure with [fare_rules.t
 |  `destination_id` | Foreign ID referencing `stops.zone_id` | Optional | Identifies a destination zone. If a fare class has multiple destination zones, create a record in [fare_rules.txt](#fare_rules.txt) for each `destination_id`.<hr>*Example: The `origin_id` and `destination_id` fields could be used together to specify that fare class "b" is valid for travel between zones 3 and 4, and for travel between zones 3 and 5, the [fare_rules.txt](#fare_rules.txt) file would contain these records for the fare class:* <br>`fare_id,...,origin_id,destination_id` <br>`b,...,3,4`<br> `b,...,3,5` |
 |  `contains_id` | Foreign ID referencing `stops.zone_id` | Optional | Identifies the zones that a rider will enter while using a given fare class. Used in some systems to calculate correct fare class. <hr>*Example: If fare class "c" is associated with all travel on the GRT route that passes through zones 5, 6, and 7 the [fare_rules.txt](#fare_rules.txt) would contain these records:* <br> `fare_id,route_id,...,contains_id` <br>  `c,GRT,...,5` <br>`c,GRT,...,6` <br>`c,GRT,...,7` <br> *Because all `contains_id` zones must be matched for the fare to apply, an itinerary that passes through zones 5 and 6 but not zone 7 would not have fare class "c". For more detail, see [https://code.google.com/p/googletransitdatafeed/wiki/FareExamples](https://code.google.com/p/googletransitdatafeed/wiki/FareExamples) in the GoogleTransitDataFeed project wiki.* |
 
+### fare_containers.txt
+
+To describe fare containers, which are used to load fare products such as tickets, passes, and discounted fares.
+
+File: **Optional** 
+
+Primary Key (`fare_container_id`)
+
+|  Field Name | Type | Presence | Description |
+|  ------ | ------ | ------ | ------ |
+|  `fare_container_id` | Unique ID | **Required** | Identifies a fare container. |
+|  `fare_container_name` | Text | Optional | The name of the fare container as displayed to riders. |
+|  `fare_container_type` | Enum | Optional | The type of the fare container.<br>Valid options are:<br><br>`0` - Physical card.<br>`1` - Digital card or mobile app.<br>`2` - Both.|
+|  `amount` | Non-negative currency amount | Optional | The cost to acquire the fare container. Does not include any amount that could be used as a fare.|
+|  `minimum_initial_purchase` | Non-negative currency amount | Optional | The amount of the minimum fare purchase required when acquiring the container. |
+|  `currency` | Currency code | **Conditionally Required** | The currency of `fare_containers.amount` or `fare_containers.minimum_initial_purchase`. <br/> Conditionally Required: <br/> - **Required** if `fare_containers.amount` or `fare_containers.minimum_initial_purchase` are defined. <br/> - **Forbidden** if `fare_containers.amount` and `fare_containers.minimum_initial_purchase` are empty. |
+
+
 ### fare_products.txt
 
 File: **Optional**
 
-Primary Key (`fare_product_id`)
+Primary Key (`fare_product_id`, `fare_container_id`)
```

Comment:
@npaun
It looks like we have two possible designs here, are we OK with both?

fare_product_id | fare_product_name | amount | currency
-- | -- | -- | --
presto_fare | PRESTO Card Fare | 3.2 | CAD
cash_fare | Cash Fare | 3.25 | CAD

fare_product_id | fare_product_name | amount | currency | fare_container_id
-- | -- | -- | -- | --
regular_fare | PRESTO Card Fare | 3.2 | CAD | PRESTO card
regular_fare | Cash Fare | 3.25 | CAD |  

Reply:
I don't think so. If you have presto_fare and cash_fare then you need to duplicate all the lines in all the other tables that reference fare_products. On the other hand when you have regular_fare it also makes it produce/read/understand easier and also has the additional information that the 2 options of $3.2 and $3.25 are kind of the same, and it depends if you have got the PRESTO Card or not.

Reply:
@isabelle-dr The two examples you show are both legal but they mean entirely different things. 

* In the first case, there are two separate kinds of tickets and none of them are associated with a fare container (despite the fact that one of them has "PRESTO" in its name). 
* In the second example, the `regular_fare` ticket is available both when using a PRESTO Card and when paying by cash.

If I were modelling the fare system in Toronto, I'd use the second approach, for the same reasons as @flocsy explained.

Reply:
Thanks for answering!
Very clear